tests/lint/test_pylinter.py

```diff
@@ -21,16 +22,16 @@ def raise_exception(*args: Any, **kwargs: Any) -> NoReturn:
 
 @patch.object(FileState, "iter_spurious_suppression_messages", raise_exception)
 def test_crash_in_file(
-    linter: PyLinter, capsys: CaptureFixture[str], tmpdir: LocalPath
+    linter: PyLinter, capsys: CaptureFixture[str], tmp_path: Path
 ) -> None:
     with pytest.warns(DeprecationWarning):
         args = linter.load_command_line_configuration([__file__])
-    linter.crash_file_path = str(tmpdir / "pylint-crash-%Y")
+    linter.crash_file_path = str(tmp_path / "pylint-crash-%Y")
     linter.check(args)
     out, err = capsys.readouterr()
     assert not out
     assert not err
-    files = tmpdir.listdir()  # type: ignore[no-untyped-call]
+    files = os.listdir(tmp_path)
```

Comment:
For future reference, this could also be achieved with `tmp_path.iterdir`. Seems like something a linter could warn about :smile:

Reply:
This can even be generalized : #7835 :) And in python 3.12 Path.walk() will also exists so there's going to be a lot of message to raise.